order-api/src/main/java/com/example/orderapi/global/config/KafkaConsumerConfig.java

```diff
@@ -0,0 +1,45 @@
+package com.example.orderapi.global.config;
+
+import com.example.kafka.Event;
+import io.confluent.kafka.serializers.KafkaAvroDeserializer;
+import org.apache.kafka.clients.consumer.ConsumerConfig;
+import org.apache.kafka.common.serialization.StringDeserializer;
+import org.springframework.context.annotation.Bean;
+import org.springframework.context.annotation.Configuration;
+import org.springframework.kafka.annotation.EnableKafka;
+import org.springframework.kafka.config.ConcurrentKafkaListenerContainerFactory;
+import org.springframework.kafka.core.ConsumerFactory;
+import org.springframework.kafka.core.DefaultKafkaConsumerFactory;
+
+import java.util.HashMap;
+import java.util.Map;
+
+@EnableKafka
+@Configuration
+public class KafkaConsumerConfig {
+
+	private static final String BOOTSTRAP_SERVER = "localhost:10000";
+
+	private static final String SCHEMA_REGISTRY_URL_CONFIG = "schema.registry.url";
+	private static final String SCHEMA_REGISTRY_URL = "http://localhost:9001";
+
+	@Bean
+	public ConsumerFactory<String, Event> consumerFactory() {
+		Map<String, Object> config = new HashMap<>();
+		config.put(ConsumerConfig.BOOTSTRAP_SERVERS_CONFIG, BOOTSTRAP_SERVER);
+		config.put(ConsumerConfig.KEY_DESERIALIZER_CLASS_CONFIG, StringDeserializer.class);
+		config.put(ConsumerConfig.VALUE_DESERIALIZER_CLASS_CONFIG, KafkaAvroDeserializer.class);
+		config.put(SCHEMA_REGISTRY_URL_CONFIG, SCHEMA_REGISTRY_URL);
+		config.put("specific.avro.reader", true);
+
+		return new DefaultKafkaConsumerFactory<>(config);
+	}
+
+	@Bean
+	public ConcurrentKafkaListenerContainerFactory<String, Event> kafkaListenerContainerFactory() {
+		ConcurrentKafkaListenerContainerFactory<String, Event> factory = new ConcurrentKafkaListenerContainerFactory<>();
+		factory.setConsumerFactory(consumerFactory());
```

Comment:
```suggestion
                factory.setErrorHandler(new DefaultErrorHandler());
```
에러핸들러를 추가해주는것도 좋아보입니다 ㅎㅎ
이외에도 다양한 설정들이 있으니 한번 찾아보시는것도 좋을것 같아요 ( ackmode , retry, concurrency ...)

Reply:
넵! 에러핸들러, 재시도 횟수 등 추가 할 수 있는 설정 추가 하겠습니다.